dask_jobqueue/core.py

```diff
@@ -421,9 +422,12 @@ def __init__(
                 "or SGEJob with the job_cls= argument."
             )
 
-        if config_name:
+        if config_name is not None:
+            self.config_name = config_name
```

Comment:
Not convinced this is the right fix, not 100% sure though. I need to look into it more ...

My instincts is that there is some unnecessary complicated things with config_name both a class variable at the Cluster level and the Job level ... a bit vague but I'll try to look at it tomorrow in more details.

Reply:
Yeah I agree this might thought about it a little more, but maybe we could do this after fixing this problem? (I know, we'll probably never look at it again ...)

Maybe we should merge this and open another issue to indicate that we parse the config twice and create duplicate class variables?

Reply:
Wow it has been 19 days already ... I have a WIP PR in #366. Let me know if you have any comments.